main.go

```diff
@@ -24,12 +27,22 @@ var (
 )
 
 func main() {
-	if len(os.Args) < 2 {
+	// We do our own flagset to be able to test arguments.
+	flagset := flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
+	flagset.SetOutput(stderr)
+	flagset.BoolVar(&yes, "yes", false, "Don't ask before syncing")
+	err := flagset.Parse(os.Args[1:])
+	if err != nil {
+		flagset.PrintDefaults()
+		exit(1)
+	}
+
+	if flagset.NArg() < 1 {
```

Comment:
I think all this flag setting would be more digestable in a seperate function. May move all that sanity checking below with it - while you're at it.

Reply:
Great idea. It also made testing easier.